### PR TITLE
Added aggregation functionality to MultivariateEvaluator

### PR DESCRIPTION
**Pull request by mbohlkeschneider**
*Description of changes:*

Second pull request for the multivariate model. This ports over the ability of the `MultivariateEvaluator` to evaluate on aggregates of the target and forecast (the aggregates should usually be more consistent in a multivariate model than in an univariate model). 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/evaluation/_base.py [hunk 1]
```diff
@@ -554,4 +625,21 @@ def __call__(
...
         )
 
+        if len(self.target_agg_funcs) != 0:
+            multivariate_metrics = {
+                key_value[0]: self.calculate_aggregate_multivariate_metrics(
+                    ts_iterator_set[-(index + 1)],
+                    fcst_iterator_set[-(index + 1)],
+                    key_value[1],
+                )
+                for index, key_value in enumerate(
```
Comment (rshyamsundar) [inline src/gluonts/evaluation/_base.py]: we can write `key_value` as a pair like this:
```
for index, (key, value) in enumerate(
```

File: src/gluonts/evaluation/_base.py [hunk 2]
```diff
@@ -554,4 +625,21 @@ def __call__(
             all_agg_metrics, all_metrics_per_ts
         )
 
+        if len(self.target_agg_funcs) != 0:
```
Comment (rshyamsundar) [inline src/gluonts/evaluation/_base.py]: minor: one can check if the `dict` is not empty as follows (maybe also better readable):
```
if self.target_agg_funcs:
```

File: src/gluonts/evaluation/_base.py (hunk 1 above)
Comment (rshyamsundar) [reply src/gluonts/evaluation/_base.py]: Also maybe rename `(key, value)` as (`agg_fun_name`, `agg_fun`).

File: src/gluonts/evaluation/_base.py (hunk 2 above)
Comment (mbohlkeschneider) [reply src/gluonts/evaluation/_base.py]: Nice! Thank you!

File: src/gluonts/evaluation/_base.py (hunk 1 above)
Comment (mbohlkeschneider) [reply src/gluonts/evaluation/_base.py]: Thanks! Good suggestion!

File: src/gluonts/model/forecast.py [hunk 3]
```diff
@@ -266,6 +266,19 @@ def copy_dim(self, dim: int):
         """
         raise NotImplementedError()
 
+    def copy_aggregate(self, agg_fun: Callable):
```
Comment (lostella) [inline src/gluonts/model/forecast.py]: You could annotate the return type here

File: src/gluonts/model/forecast.py [hunk 4]
```diff
@@ -399,6 +413,20 @@ def copy_dim(self, dim: int):
             info=self.info,
         )
 
+    def copy_aggregate(self, agg_fun: Callable):
```
Comment (lostella) [inline src/gluonts/model/forecast.py]: And here

File: src/gluonts/evaluation/_base.py [hunk 5]
```diff
@@ -488,29 +527,57 @@ def get_eval_dims(self, target_dimensionality: int) -> List[int]:
...
         all_agg_metrics
             dictionary with aggregate metrics of individual dimensions
         all_metrics_per_ts
-            DataFrame containing metrics for all time series of all evaluated dimensions
+            DataFrame containing metrics for all time series of all evaluated
+            dimensions
 
         Returns
         -------
-        Dict[int, Dict[str, float]]
```
Comment (lostella) [inline src/gluonts/evaluation/_base.py]: I think the type annotation can stay here, see https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html

File: src/gluonts/evaluation/_base.py [hunk 6]
```diff
@@ -488,29 +527,57 @@ def get_eval_dims(self, target_dimensionality: int) -> List[int]:
...
+            self.extract_aggregate_forecast(forecast_iterator, agg_fun),
+        )
+        return agg_metrics
+
     def calculate_aggregate_vector_metrics(
         self,
         all_agg_metrics: Dict[int, Dict[str, float]],
         all_metrics_per_ts: pd.DataFrame,
-    ):
+    ) -> Dict[int, Dict[str, float]]:
```
Comment (lostella) [inline src/gluonts/evaluation/_base.py]: I’m wondering if the gluonts.evaluation makes mypy happy? You could run mypy against it, and if everything is fine mark it as typesafe by adding the `.typesafe` file into it

Comment (mbohlkeschneider) [reply src/gluonts/evaluation/_base.py]: Yeah, the typing of this module is a bit messy. I'll refactor a bit and clean it up. Thank you for the comment!

**Pull request merged**